server/controllers/notification.js

```diff
@@ -0,0 +1,70 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route PUT /profile/edit
+// @desc edit user profile
+// @access Public
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+const userId = await User.findById(req.user.id);
+const { type, title, description } = req.body;
+const notification = await Notification.create({userId, type, title, description});
+
+  res.status(200).json({
+    success: {
+      notification: notification
+    },
+  });
+
+});
+
+
+exports.markAsRead = asyncHandler(async (req, res, next) => {
+const {notificationId} = req.params;
+const update = { read: true };
+const foundNotification = await Notification.findOneAndUpdate(notificationId, update);
+const readNotification = await foundNotification.save()
+
+  res.status(200).json({
+    success: {
+      notification: readNotification
+    },
+  });
+
+});
+
+exports.retrieveUsersNotifications = asyncHandler(async (req, res, next) => {
+const userId = req.user.id;
+const foundNotifications = await Notification.find({ userId: userId })
+
+    if (!foundNotifications) {
+    res.status(404);
+    throw new Error("No notifications found");
+  }
+
+    res.status(200).json({
+    success: {
+      notifications: foundNotifications,
+    },
+  });
+
+});
+
+exports.retrieveUsersUnreadNotifications = asyncHandler(async (req, res, next) => {
+const userId = req.user.id;
+const foundNotifications =  await Notification.find({ userId: userId, read: false })
+
+    if (!foundNotifications) {
+    res.status(404);
+    throw new Error("No notifications found");
+  }
```

Comment:
Similar comments here 👍🏼

Reply:
Okay I thought 404 meant not found, not that it meant we could not located the resource, so thanks for clarifying this

Reply:
Just to clarify, [`404`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404?retiredLocale=id) *does* mean not found. But finding an empty collection still counts as finding the resource, it being empty shouldn't lead to a `404`. Once again, [this](https://stackoverflow.com/a/13367198/15220729) stackoverflow answer details the reasoning behind this 👍🏼